### PR TITLE
[Shop] Strip _locale parameter from URL if using storage locale switcher

### LINKED ISSUE #7970: Disabling localised URLs passes _locale as query string
I've followed the steps from http://docs.sylius.org/en/latest/cookbook/disabling-localised-urls.html but now the URLs are generated with the _locale passed in the query string (?locale=en_US). Is there a way to disable that?

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | kinda |
| New feature?    | kinda |
| BC breaks?      | a really small one |
| Related tickets | fixes #7970 |
| License         | MIT |

### CODE REVIEW

**Comment by stefandoorn** [comment]
Looks good to me. Using a similar one in production, although mine is slightly different as I'm also using the CmsPlugin, therefore I'm decorating CmfRouter. Hopefully these two play well together.. :-)

**Comment by pjedrzejewski** [comment]
@pamil Let's fix the conflict in the UPGRADE.md file and it is good to go.

File: src/Sylius/Bundle/ShopBundle/spec/Router/LocaleStrippingRouterSpec.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    function it_strips_locale_from_the_generated_url_if_locale_is_the_same_as_the_one_from_context(
+        RouterInterface $decoratedRouter,
+        LocaleContextInterface $localeContext
+    ): void {
+        $localeContext->getLocaleCode()->willReturn('pl_PL');
+
+        $decoratedRouter
+            ->generate('route_name', [], UrlGeneratorInterface::ABSOLUTE_PATH)
+            ->willReturn('http://generated.url/?_locale=pl_PL')
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/Router/LocaleStrippingRouterSpec.php]: Can we test for each case? It should be as easy as adding here two more examples and calling generate two more times

Comment (pamil) [reply src/Sylius/Bundle/ShopBundle/spec/Router/LocaleStrippingRouterSpec.php]: Done! 👌

**Comment by pjedrzejewski** [comment]
Thanks Kamil! 👍

**Pull request merged**